### PR TITLE
fix lpad/rpad trailing zero miss

### LINKED ISSUE #9465: `lpad/rpad` return wrong result if it need return null for not null input
## Bug Report

Please answer these questions before submitting your issue. Thanks!

In https://github.com/pingcap/tiflash/blob/78bd3f04dc5f10b94aef13cf3d9e0d42a4c808e5/dbms/src/Functions/FunctionsString.cpp#L3588-L3591
And 
https://github.com/pingcap/tiflash/blob/78bd3f04dc5f10b94aef13cf3d9e0d42a4c808e5/dbms/src/Functions/FunctionsString.cpp#L3662-L3665

It returns `true`(which means the result is `null`) without adding the tailing `\0`

### PR DESCRIPTION
**Pull request by guo-shaoge**
### What problem does this PR solve?

Issue Number: close #9465

Problem Summary:

### What is changed and how it works?

```commit-message

```

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
fix lpad/rpad trailing zero miss
```

### CODE REVIEW

File: tests/fullstack-test/expr/pad.test [hunk 1]
```diff
@@ -111,3 +111,19 @@ mysql> set tidb_isolation_read_engines='tiflash'; set tidb_enforce_mpp=1; select
...
+
+
+mysql> drop table if exists test.t1
+mysql> create table test.t1(c1 varchar(100), c2 int)
+mysql> alter table test.t1 set tiflash replica 1
+mysql> insert into test.t1 values('a', -1)
+func> wait_table test t1
+
+# crc32 will call ColumnString::getDataAt(i), which assume all string rows ends with '\0'.
+mysql> select crc32(lpad(c1, c2, 'b')) from test.t1
```
Comment (yibin87) [inline tests/fullstack-test/expr/pad.test:123]: add rpad cases also?

File: dbms/src/Functions/FunctionsString.cpp [hunk 2]
```diff
@@ -3622,6 +3622,13 @@ class TidbPadImpl
         }
     }
 
+    static void addTrailingZero(ColumnString::Chars_t & res, ColumnString::Offset & res_offset)
+    {
+        res.resize(res.size() + 1);
+        res[res_offset] = '\0';
+        res_offset++;
```
Comment (yibin87) [inline dbms/src/Functions/FunctionsString.cpp:3629]:
```suggestion
        ++res_offset;
```

File: tests/fullstack-test/expr/pad.test (hunk 1 above)
Comment (guo-shaoge) [reply tests/fullstack-test/expr/pad.test:123]: done

File: dbms/src/Functions/FunctionsString.cpp (hunk 2 above)
Comment (guo-shaoge) [reply dbms/src/Functions/FunctionsString.cpp]: done

File: dbms/src/Functions/FunctionsString.cpp [hunk 3]
```diff
@@ -3622,6 +3622,13 @@ class TidbPadImpl
         }
     }
 
+    static void addTrailingZero(ColumnString::Chars_t & res, ColumnString::Offset & res_offset)
```
Comment (windtalker) [inline dbms/src/Functions/FunctionsString.cpp:3625]: L3616-L3618 can also use `addTrailingZero` instead.

**Review by windtalker — APPROVED** [review]
lgtm

**Review by yibin87 — APPROVED** [review]
LGTM

**Pull request merged**